homeassistant/components/button/__init__.py

```diff
@@ -141,7 +142,7 @@ async def async_internal_added_to_hass(self) -> None:
         """Call when the button is added to hass."""
         await super().async_internal_added_to_hass()
         state = await self.async_get_last_state()
-        if state is not None and state.state is not None:
+        if state is not None and state.state not in (STATE_UNAVAILABLE, None):
```

Comment:
what about `STATE_UNKNOWN` ?

Reply:
unknown is the default state if we don't have a timestamp so I think its probably fine to restore it